### ISSUE TITLE
[QUESTION] Best method to install a fresh instance of Learning Pathways

### ISSUE THREAD

**Issue by DaveTheTeamsGuy** [pnp/custom-learning-office-365#641]
### What is the context of your question

- [x] Installation
- [ ] Learning Pathways Content
- [ ] SharePoint Success Content
- [ ] Administration
- [ ] Custom Content
- [ ] Viewer Web Part

### What is your question
What is the best way to install a fresh, out of the box instance of learning pathways (non-lookbook) when we have a previous version that was installed and customized years ago by another group?  Is it acceptable to just use the [steps to provision a new solution](https://docs.microsoft.com/en-us/office365/customlearning/custom_provision), choose a different URL, then use the Set-PnPStorageEntity cmdlet to make the new instance the master?  Or can we just use the instructions above to reinstall over the existing solution with the expectation that it will reset all of the customizations?

Thank you - 

>Provide related screen-shots to help us understand your question

**Comment by v-saikirang** [comment]
Hi @DaveTheTeamsGuy,

Thanks for your interest in the app. Please allow us some time. We will get back to you ASAP.

Thanks.

**Comment by dcashpeterson** [comment]
@DaveTheTeamsGuy the recommended approach if you already have learning pathways installed is to do an [upgrade ](https://docs.microsoft.com/en-us/office365/customlearning/custom_update) as this will keep any customizations you have previously made. If you just want to start fresh my suggestion would be to uninstall the previous version of the sppkg file from the app catalog and recycle bin and delete the site collection (including emptying the tenant site collection app catalog). Then either follow the manual steps you outlined above or use the LookBook.

**Comment by DaveTheTeamsGuy** [comment]
Thanks @dcashpeterson.  We need to maintain some of the old custom pages until the new instance is up.  Thoughts on this process?

1. Pull the sppkg file from the app catalog.
2. Install a new instance using https://docs.microsoft.com/en-us/office365/customlearning/custom_provision
3. Dumpster the old site collection.

Outstanding question is, will we need to run the SPO PnP cmdlet I referenced, or should the instructions above switch to the new master?

**Comment by dcashpeterson** [comment]
@DaveTheTeamsGuy If you need to keep some of the old data then I would follow the upgrade path I linked to earlier. That is the easiest and safest way to do it. That way you don't loose data. Depending on the version you are upgrading from you might need to run the update function but that gets triggered from the admin screen. I would not follow the steps you mentioned above if you need to keep data. 

Just to be clear if you change the URL of the site collection for Learning Pathways you need to run that PowerShell command to add/update the tenant app property. We have a PowerShell script that does that for you [here](https://github.com/pnp/custom-learning-office-365/blob/main/installation/UpdateM365lpSiteUrl.ps1).

**Comment by DaveTheTeamsGuy** [comment]
We only need to keep one page worth of links, so I think a fresh install is the way to go.  Unfortunately I attempted a new install and got a SharePoint provisioning service error while invoking a graph endpoint.

**Comment by DaveTheTeamsGuy** [comment]
We got everything running.  It was pretty simple, actually.  The tenant admin account I was using needed admin at our SPO root site, then we discovered that the old learning pathways app was checked out.  Once both of those were resolved, the lookbook install created the new instance, updated the existing app and everything including the old site is working.  We now just need to move our customizations over to the new site, change some links and dumpster the old site.  Thanks again for the tips and advice.
